### ISSUE TITLE
Import block inexplicably raises error for some attributes

### ISSUE THREAD

**Issue by nils-werner** [gnuradio/gnuradio#5783]
### What happened?

With a file `examples/importme.py` 

```python
a = 5

def b():
    return 10

def d():
    return 10
```

and using an import block the following imports seem to work

```python
from examples.importme import a
from examples.importme import b
```

and the following statement does not work

```python
from examples.importme import d
```

I would assume that all statements should succeed.

### System Information

OS: Arch Linux
GR Installation Method: Conda

### GNU Radio Version

3.10 (maint-3.10)

### Specific Version

3.10.1.1

### Steps to Reproduce the Problem

1. Create python file
2. Add import block
3. Try mentioned combinations

**Comment by nils-werner** [comment]
I think I figured it out: GRC does the import only once, and whatever was defined at that time is assumed to be there forever. If I change a file that I am importing, I need to restart GRC for the new names to become valid.

**Comment by marcusmueller** [comment]
> I think I figured it out: GRC does the import only once, and whatever was defined at that time is assumed to be there forever.

Nope, that's the Python interepreter's `import` semantics! GRC does nothing special on top of that.

> If I change a file that I am importing, I need to restart GRC for the new names to become valid.

That now is a bit suboptimal; can you confirm that closing and opening the same flow graph file does not suffice, and neither does clicking the "reload library" button?

**Comment by BStudent** [comment]
can confirm.   i'm accustomed to python import semantics, I doubt that's what's going on.  
it looks more like what happens with an intransigent __pycache__, which would be solvable by deleting the __pycache__, except that I see no no such file in my relevant paths.
I'm using radioconda on Mac Arm64, but have had this happen in the past in Linux as well (w/o radioconda).   

In the full module shown below, (FTR - named grzdaq.py and in the right path, etc) I get the following behavior: 
1. the (all-caps/dunder) vars import consistently, as does `chan_address(...)`.  
2. p_tag_kv() never imports (i.e. "failed" with no explanation). it imports fine in IDE and console.
3. `foo_bar()` imporrs sometimes, sometimes not.   
4.  `baz_barf()` imported once, with an extra param.  
**Behavior is independent of**: (A) order of import, and (B) order of declaration.
I've tried restarting grc, refreshing blocks, and everything except a cold reboot.
The code below has also been through too many permutations to count with regard to using/not-using default values, Optional[] params (with back-filling None values), etc.  

```python
from typing import Optional

__IP_ADDR_DEFAULT__: str = '127.0.0.1'
CHANNEL_1_PORT: int = 23661
IP_ADDR_ALL: str = __IP_ADDR_DEFAULT__

def chan_address(CHAN: int, IP_ADDR: Optional[str] = None, IP_use_config: Optional[str] = None) -> str:

    if IP_ADDR is None:
        if IP_use_config is None: # TODO make a runconfig option for stuff like this
            IP_ADDR: str = IP_ADDR_ALL
            ...
        else: # HACK: don't assert/raise here: just use same as above for now ...
            IP_ADDR: str = __IP_ADDR_DEFAULT__
            ...
        ...

    if CHAN < 1:
        print(f"WARNING: CHAN should be >= 1, got {CHAN} USING CHAN=1 instead.") # , file=sys.stderr)
        CHAN=1
        ...

    port: int = (CHANNEL_1_PORT - 1) + CHAN

    return f"tcp://{IP_ADDR}:{port}"

def p_tag_kv(chan,
             tcode,
             vlen = 1,
             SAMP_RATE = 32000 ) -> str:
    p_tag: str = f"CHAN={chan}__TCODE={tcode}__VLEN={vlen}__samp_rate={SAMP_RATE}"   #__PLEN={plen}"
    return p_tag

def foo_bar(plen: int) -> str:
    return f"PLEN={plen}"

def baz_barf(chan = 1) -> str:
    p_tag: str = f"chan={chan}" 
    return p_tag
```

**Comment by BStudent** [comment]
## ISSUE w/ GRC 'import' block: An Improved Example, with Experiment

### SETUP: here's a slightly cleaned-up version of example code above:

```python
from typing import Optional

__IP_ADDR_DEFAULT__: str = '127.0.0.1'
CHANNEL_1_PORT: int = 23661
IP_ADDR_ALL: str = __IP_ADDR_DEFAULT__

def chan_address(CHAN: int, IP_ADDR: Optional[str] = None, IP_use_config: Optional[str] = None) -> str:
    if IP_ADDR is None:
        if IP_use_config is None: # NOTE: config not yet impl, and this code has devolved into bug-demo anyway ...
            IP_ADDR: str = IP_ADDR_ALL
            ...
        else: # HACK: (DO NOT RAISE fake it w/ __IP_ADDR_DEFAULT__ ... (good for demo purposes)
            IP_ADDR: str = __IP_ADDR_DEFAULT__
            ...
        ...
    if CHAN < 1:  # DO NOT RAISE, just issue a sternly-worded message ...
        print(f"WARNING: CHAN should be >= 1, got {CHAN} USING CHAN=1 instead.") # , file=sys.stderr)
        CHAN=1
        ...
    port: int = (CHANNEL_1_PORT - 1) + CHAN
    return f"tcp://{IP_ADDR}:{port}"

def p_tag_kv(CHAN,
             vlen,
             plen,
             SAMP_RATE) -> str:
    p_tag: str = f"CHAN={CHAN}_VLEN={vlen}__samp_rate={SAMP_RATE}__PLEN={plen}"
    return p_tag
```

## What goes wrong? Issue and Steps to Reproduce:

1. In the GRC UI, create an .`**import**` block (its name doesn't matter, within reason)
2. Paste this line into the block:
`from grzdaq import chan_address, CHANNEL_1_PORT, __IP_ADDR_DEFAULT__, IP_ADDR_ALL`

#### RESULT ... that works. Other GR Blocks can access the function and vars.  NEXT ...

3. Paste **THIS** code in the import statement instead...
`from grzdaq import chan_address, CHANNEL_1_PORT, __IP_ADDR_DEFAULT__, IP_ADDR_ALL, p_tag_kv`

#### RESULT... RED TEXT!  A message is provided (in block and in vars window) proclaiming failure.

> Param - Import(imports):
>    Import "from grzdaq import chan_address, CHANNEL_1_PORT, __IP_ADDR_DEFAULT__, IP_ADDR_ALL, p_tag_kv" failed.

#### NOTES:
  1. same issue with/without GRC restart, sys reboot, reinstall of GRC, trying a different computer, etc.
  2. order of import or declaration doesn't matter.  neither does renaming the function etc.

#### WHAT's wrong with the 'p_tag_kv' function that forbids its participation in GR??  
- WE do not know.
- _Perhaps the function itself is **toxic** for reasons we cannot fathom, and GRC is defending us from its harmful effects?_

#### THAT'S TESTABLE
We can visit the alternate universe where GRC allows us to enter _"the forbidden import statement"_ by editing the
auto-generated python code: what will happen?  Let's find out.

**Comment by BStudent** [comment]
@marcusmueller re the above comments / excerpts, in my case this could be a 'perfect storm' of interactions - stuff specific to my code combined with a couple of 'gotchas' in block-to-python conversion.  
I'm working on a project that (as a side-effect) might provide some helpful examples / tutorials / clarifications for using python in GR/GRC.  Happy to help.  
I don't use this account much - I'll sign up for the gr wiki page from the account for that project when I get it cleaned up.   Thanks for all your hard work!
